### PR TITLE
Update ring docs for event platform

### PR DESCRIPTION
**Pull request by sdb9696**
## Proposed change

Update the ring docs to reflect the new `event` platform and the pending deprecation of the `binary_sensor` platform.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/125506
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Enhanced Ring integration documentation with a detailed description of the new event entity.
	- Introduced a directive for users to transition from the binary sensor to the new event entity, expanding integration capabilities.
	- Specified responses for the binary sensor, including motion detection, doorbell rings, and intercom unlock events.

### CODE REVIEW

File: source/_integrations/ring.markdown [hunk 1]
```diff
@@ -49,7 +51,9 @@ This integration does NOT allow for live viewing of your Ring camera within Home
 
 ## Binary sensor
 
-Once you have enabled the [Ring integration](/integrations/ring), you can start using a binary sensor. Currently, it supports doorbell, external chimes and stickup cameras.
+The binary sensor switches between off and on when motion, doorbell rings, and intercom unlock events occur.
+
+The binary sensor is being replaced with the event entity and you should migrate any automations to the event entity by release 2025.3.0.
```
Comment (bdraco) [inline source/_integrations/ring.markdown]: We should add a followup PR to add a repair issue to notify them about this and tell them which automations/scripts are using it.

example to find them https://github.com/home-assistant/core/blob/20600123f8b2f6fabe4d89ae866bc467dc24c04d/homeassistant/components/unifiprotect/migrate.py#L40

File: source/_integrations/ring.markdown [hunk 2]
```diff
@@ -49,7 +51,9 @@ This integration does NOT allow for live viewing of your Ring camera within Home
 
 ## Binary sensor
 
-Once you have enabled the [Ring integration](/integrations/ring), you can start using a binary sensor. Currently, it supports doorbell, external chimes and stickup cameras.
+The binary sensor switches between off and on when motion, doorbell rings, and intercom unlock events occur.
```
Comment (bdraco) [inline source/_integrations/ring.markdown]:
```suggestion
The binary sensor switches off and on when motion, doorbell rings, and intercom unlock events occur.
```

File: source/_integrations/ring.markdown (hunk 1 above)
Comment (bdraco) [inline source/_integrations/ring.markdown]:
```suggestion
The binary sensor is being replaced with the event entity, and you should migrate any automations to the event entity by release 2025.3.0.
```

Comment (bdraco) [reply source/_integrations/ring.markdown]: Looks like its already implemented in the other PR https://github.com/home-assistant/core/blob/dev/homeassistant/components/ring/entity.py#L41

File: source/_integrations/ring.markdown (hunk 2 above)
Comment (sdb9696) [reply source/_integrations/ring.markdown]: Done

File: source/_integrations/ring.markdown (hunk 1 above)
Comment (sdb9696) [reply source/_integrations/ring.markdown]: Done and corrected the target date to 2025.4.0

**Review by bdraco — APPROVED** [review]

**Review by bdraco — APPROVED** [review]

**Pull request merged**